massa-ledger-worker/src/ledger_db.rs

```diff
@@ -210,6 +197,162 @@ impl LedgerDB {
         self.write_batch(batch);
     }
 
+    /// Get the current disk ledger hash
+    pub fn get_ledger_hash(&self) -> Hash {
+        let handle = self.db.cf_handle(METADATA_CF).expect(CF_ERROR);
+        if let Some(ledger_hash_bytes) = self
+            .db
+            .get_pinned_cf(handle, LEDGER_HASH_KEY)
+            .expect(CRUD_ERROR)
+            .as_deref()
+        {
+            Hash::from_bytes(ledger_hash_bytes.try_into().expect(LEDGER_HASH_ERROR))
+        } else {
+            // initial ledger_hash value to avoid matching an option in every XOR operation
+            // because of a one time case being an empty ledger
+            // also note that the if you XOR a hash with itself result is LEDGER_HASH_INITIAL_BYTES
+            Hash::from_bytes(LEDGER_HASH_INITIAL_BYTES)
+        }
+    }
+
+    /// Get the given sub-entry of a given address.
+    ///
+    /// # Arguments
+    /// * `addr`: associated address
+    /// * `ty`: type of the queried sub-entry
+    ///
+    /// # Returns
+    /// An Option of the sub-entry value as bytes
+    pub fn get_sub_entry(&self, addr: &Address, ty: LedgerSubEntry) -> Option<Vec<u8>> {
+        let handle = self.db.cf_handle(LEDGER_CF).expect(CF_ERROR);
+        self.db
+            .get_cf(handle, ty.derive_key(addr))
+            .expect(CRUD_ERROR)
+    }
+
+    /// Get every key of the datastore for a given address.
+    ///
+    /// # Returns
+    /// A `BTreeSet` of the datastore keys
+    pub fn get_datastore_keys(&self, addr: &Address) -> Option<BTreeSet<Vec<u8>>> {
+        let handle = self.db.cf_handle(LEDGER_CF).expect(CF_ERROR);
+
+        let mut opt = ReadOptions::default();
+        opt.set_iterate_range(data_prefix!(addr).clone()..end_prefix(data_prefix!(addr)).unwrap());
+
+        let mut iter = self
+            .db
+            .iterator_cf_opt(handle, opt, IteratorMode::Start)
+            .flatten()
+            .map(|(key, _)| key.split_at(ADDRESS_SIZE_BYTES + 1).1.to_vec())
+            .peekable();
+
+        // Return None if emtpty
```

Comment:
```suggestion
        // Return None if empty
        // TODO: function should return None if complete entry does not exist
        // and Some([]) if it does but datastore is empty
```

Reply:
I've corrected my mistake of only changing the typo.